packages/open-response/src/components/runtime.tsx

```diff
@@ -110,13 +111,19 @@ export const Runtime: React.FC<IProps> = ({ authoredState, interactiveState, set
     voiceTypingRef.current = voiceTypingRef.current || new VoiceTyping();
     if (voiceTypingActive) {
       initialTextRef.current = textAreaRef.current?.value || "";
+      lastTranscriptRef.current = "";
       voiceTypingRef.current.enable(transcript => {
-        const selection = selectionRef.current || {start: 0, end: 0};
-        handleUpdateTextArea([
-          initialTextRef.current.substring(0, selection.start),
-          transcript,
-          initialTextRef.current.substring(selection.end)
-        ].join(""));
+        if (transcript !== lastTranscriptRef.current) {
+          const selection = selectionRef.current || {start: 0, end: 0};
+          const updatedText = [
+            initialTextRef.current.substring(0, selection.start),
+            transcript,
+            initialTextRef.current.substring(selection.end)
+          ].join("");
+          handleUpdateTextArea(updatedText);
+          lastTranscriptRef.current = transcript;
+          log("voice typing updated", {voiceText: transcript, updatedText});
```

Comment:
If I understand this right, it seems like it will be a lot of log message each time the text area is updated. That seems like a lot of log messages to record. I'm pretty sure when the user is typing we buffer this a bit and don't record every change to the text area.

But if this is what Dan needs, then it is fine.